### PR TITLE
wot: Web-of-Trust attestation proof chainlinks

### PR DESCRIPTION
**Pull request by maxtaco**

### CODE REVIEW

File: src/wot.iced [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        schema.dict({domain : schema.string(), protocol : schema.string() })
+      ])
+      state : schema.int()
+      proof_type : schema.int()
+    })
+
+    schm = schema.dict({
+      user : schema.dict({
+        username : schema.string()
+        uid : schema.uid().convert()
```
Comment (heronhaye) [inline src/wot.iced]: do we need this? (if the goal is to save space)

File: src/wot.iced [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      })
+      confidence : schema.dict({
+        proofs : schema.array(proof_schema).optional()
+        vouched_by : schema.array(schema.uid().convert()).optional()
+        keybase_username : schema.string_enum(["audio","video","email","other_chat","in_person"]).optional()
+        other : schema.string().optional()
+        keybase_history : schema.int().optional()
+      })
+      failing_proofs : schema.array(proof_schema).optional()
+      attestation : schema.array(schema.string())
```
Comment (heronhaye) [inline src/wot.iced]: what is this?

File: src/wot.iced [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        username : schema.string()
+        uid : schema.uid().convert()
+        eldest_seqno : schema.int()
+        last_seqno : schema.int()
+        eldest_kid : schema.kid().convert()
+      })
+      confidence : schema.dict({
+        proofs : schema.array(proof_schema).optional()
+        vouched_by : schema.array(schema.uid().convert()).optional()
+        keybase_username : schema.string_enum(["audio","video","email","other_chat","in_person"]).optional()
```
Comment (heronhaye) [inline src/wot.iced]: maybe "username_verified_via"?

also, why string enums here and below but integer enums everywhere else?

File: src/wot.iced [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    schm = schema.dict({
+      user : schema.dict({
+        username : schema.string()
+        uid : schema.uid().convert()
+        eldest_seqno : schema.int()
+        last_seqno : schema.int()
+        eldest_kid : schema.kid().convert()
+      })
+      confidence : schema.dict({
+        proofs : schema.array(proof_schema).optional()
```
Comment (heronhaye) [inline src/wot.iced]: if these don't check out, will it silently disappear from a 3rd party's view or will it be in red in the GUI?

File: src/wot.iced [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+        vouched_by : schema.array(schema.uid().convert()).optional()
+        keybase_username : schema.string_enum(["audio","video","email","other_chat","in_person"]).optional()
+        other : schema.string().optional()
+        keybase_history : schema.int().optional()
```
Comment (heronhaye) [inline src/wot.iced]: what's this?

File: src/wot.iced [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+    schm = schema.dict({
+      user : schema.dict({
+        username : schema.string()
+        uid : schema.uid().convert()
+        eldest_seqno : schema.int()
+        last_seqno : schema.int()
+        eldest_kid : schema.kid().convert()
+      })
+      confidence : schema.dict({
```
Comment (heronhaye) [inline src/wot.iced]: I wonder if we should freeze or give an option to include "full name", since that is server trust

File: src/wot.iced (hunk 1 above)
Comment (maxtaco) [reply src/wot.iced]: I guess we can do with one or the other...  Hmmm, but then which?

File: src/wot.iced (hunk 2 above)
Comment (maxtaco) [reply src/wot.iced]: A written description of how Alice knows Bob. See ticket.

File: src/wot.iced (hunk 4 above)
Comment (maxtaco) [reply src/wot.iced]: I think nothing. It *was* true at the time of the attestation, and that is all that matter for this, I think.

File: src/wot.iced (hunk 5 above)
Comment (maxtaco) [reply src/wot.iced]: How long you have known them on keybase for (just if you know them only by keybase name).  In days maybe?

File: src/wot.iced (hunk 3 above)
Comment (maxtaco) [reply src/wot.iced]: Regret for earlier mistake, I think... We can keep it with ints to keep it consistent thought...

File: src/wot.iced (hunk 6 above)
Comment (maxtaco) [reply src/wot.iced]: Good question, let's ask Chris. And bio picture too?

Comment (maxtaco) [reply src/wot.iced]: Chris says let's not for now

**Pull request merged**